### ISSUE TITLE
[BUG] The regularization_loss and total_loss are not reported anymore after moving the loss to model.compile()

### ISSUE THREAD

**Issue by gabrielspmoreira** [NVIDIA-Merlin/models#495]
### Bug description
After the loss were moved to `model.compile()`, the `regularization_loss` and `total_loss` are not reported anymore.
From the experiments results it is also not clear if the regularization losses are being summed to the final loss or not

It can be seen in the plots below that the `regularization_loss` and `total_loss` were reported before (blue line) but are not being reported after the refactory (pink line) and the loss of the new `run` is worse than before. 

![image](https://user-images.githubusercontent.com/504752/172187166-9d8aad52-e8d0-4368-a495-bd753b433584.png)

The results for the new run can be reproduced using the [retrieval experiments script](https://github.com/NVIDIA-Merlin/research/blob/main/retrieval_exp/scripts/retrieval_train_eval.py), with the following arguments and LastFM:

```bash
python scripts/retrieval_train_eval.py --dataset lastfmB --wandb_exp_group lastfmB_youtubednn_xe_sampledsofmax_v07.2 --model_type youtubednn --two_tower_activation selu --epochs 20 --lr_decay_steps 50 --output_path /results --data_path /data --log_to_wandb --optimizer adam --eval_batch_size 2048 --train_metrics_steps 100 --topk_metrics_cutoffs 100 --max_seq_length 20 --youtubednn_sampled_softmax True --fail_if_recall_at_100_higher_than 0.5 --xe_label_smoothing 0.0 --two_tower_mlp_layers None --two_tower_dropout 0.3 --two_tower_embedding_sizes_multiplier 5.0 --logits_temperature 0.8 --lr 0.02238982864512884 --lr_decay_rate 0.9400000000000001 --l2_reg 1.1472035643715902e-05 --embeddings_l2_reg 1.01774316277423e-06 --train_batch_size 4096 --item_id_emb_size 512 --youtubednn_sampled_softmax_n_candidates 500
```
